lib/SPIRV/SPIRVLowerBitCastToNonStandardType.cpp

```diff
@@ -68,15 +70,13 @@ static VectorType *getVectorType(Type *Ty) {
 /// instructions have been encountered in input LLVM IR.
 bool lowerBitCastToNonStdVec(Instruction *Inst,
                              std::vector<Instruction *> &InstsToErase) {
-  Instruction *OldInstIter(Inst);
-  IRBuilder<> Builder(OldInstIter);
-  Value *NewValue = OldInstIter->getOperand(0);
+  IRBuilder<> Builder(Inst);
+  Value *NewValue = Inst->getOperand(0);
   VectorType *NewVecTy = getVectorType(NewValue->getType());
-  InstsToErase.push_back(OldInstIter);
+  InstsToErase.push_back(Inst);
   // Handle addrspacecast instruction after bitcast if present
-  // TODO: There are can be many addrspacecast instructions using this bitcast,
-  // so that it is necessary to handle all these instructions in the same way
-  for (auto *U : OldInstIter->users()) {
+  InstructionValueVec ASCastInsts;
```

Comment:
I'd suggest to rewrite this function to call it recursively for each next instruction in the pattern.